src/TokenGenerator/PasswordGenerator.js

```diff
@@ -38,6 +39,26 @@ class PasswordGenerator extends AbstractTokenGenerator {
     return this._doFetch(parameters).then(response => response.json());
   }
 
+  _manageBadRequest(response) {
+    return response
+      .json()
+      .then(body => {
+        if (body.error === 'invalid_grant') {
+          // bad params like wrong scopes sent to oauth server
+          // will generate a 400, we want final clients to consider it
+          // like 401 in order to take proper action
+          throw new UnauthorizedError(body.error, response);
+        }
+        return handleBadResponse(response);
+      })
+      .catch(err => {
+        if (err.type === 'invalid-json') {
```

Comment:
I do not find any info in the [fetch standard](https://fetch.spec.whatwg.org/). Am I missing something ?

Reply:
This is what happens when you send a null body with jest-fecth-mock but maybe you're right and it never happens with fetch